### PR TITLE
[WIP] Slack integration

### PR DESCRIPTION
**Pull request by aileen**
ref #6584
- adding apps in navigation menu
- adding a static page for apps and a nested page for slack
- adding templates and controllers for apps & slack
- adding slack.js to send webhook to slack for new posts
- adding slack_spec.js to unit tests

Todos:
- [x] New Apps and Slack page (Slack is a nested page)
- [x] Adding slack as a settings value
- [x] coding a ping-service, which send slack a link to the post blog as soon as it's published and slack is configured inkl. unit test
- [x] adding a new api endpoint and a service to send test notifications inkl. unit tests

### CODE REVIEW

File: core/client/app/controllers/settings/apps.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+});
+export default Controller.extend(SettingsSaveMixin, {
+    slackController: controller('settings.apps.slack'),
+    newSlackConfig: null,
+
+    slack: computed('model.slack', function () {
+        let slackItem;
+
+        try {
+            [ slackItem ] = JSON.parse(this.get('model.slack') || [{}]);
```
Comment (kevinansfield) [inline core/client/app/controllers/settings/apps.js]: I think the `)` is in the wrong place here, perhaps it should be this?

```javascript
[ slackItem ] = JSON.parse(this.get('model.slack')) || [{}];
```

Comment (acburdine) [inline core/client/app/controllers/settings/apps.js]: Either that or the `|| [{}]` can be removed entirely as any parse errors would be caught by the catch statement.

Comment (kevinansfield) [inline core/client/app/controllers/settings/apps.js]: Oops, maybe it probably be something more like the following as it's there to guard against a blank string?

```javascript
[ slackItem ] = JSON.parse(this.get('model.slack') || '[{}]');
```

**Comment by ErisDS** [comment]
Closing in favour of #6795

**Pull request closed without merging**